### PR TITLE
boards : nucleof412zg - CAN Internal controller

### PR DESCRIPTION
**Pull request by pbonazzi**
Signed-off-by: Pietro Bonazzi <pietrobonazzi.open@gmail.com>

### CODE REVIEW

File: boards/arm/nucleo_f412zg/nucleo_f412zg_defconfig [hunk 1]
```diff
@@ -11,15 +11,19 @@ CONFIG_HW_STACK_PROTECTION=y
...
 
 # Enable Clocks
 CONFIG_CLOCK_CONTROL=y
 
-# enable pin controller
+# Enable pin controller
 CONFIG_PINCTRL=y
+
+# Enable CAN
+CONFIG_CAN=y
```
Comment (gmarull) [inline boards/arm/nucleo_f412zg/nucleo_f412zg_defconfig:28]: this is mixing style changes which are not related. CAN is not a minimal peripheral, it should not be in board defconfig.

**Review by gmarull — CHANGES_REQUESTED** [review]
Thanks for your contribution, please, split into at least 2 commits: one that defines new can device, 2nd for board modifications.

FYI https://static.sched.com/hosted_files/zephyr2022/a3/HOWTO_Get_Your_Zephyr_Patches_Merged_ZDS_2022.pdf

**Comment by pbonazzi** [comment]
> Thanks for your contribution, please, split into at least 2 commits: one that defines new can device, 2nd for board modifications.
> 
> FYI https://static.sched.com/hosted_files/zephyr2022/a3/HOWTO_Get_Your_Zephyr_Patches_Merged_ZDS_2022.pdf

Thank you for your review . I have implemented the changes you requested.

File: boards/arm/nucleo_f412zg/nucleo_f412zg_defconfig (hunk 1 above)
Comment (pbonazzi) [reply boards/arm/nucleo_f412zg/nucleo_f412zg_defconfig]: Thank you for your review . I have implemented the changes you requested.

**Comment by erwango** [comment]
Can you fix the issues reported here:
https://github.com/zephyrproject-rtos/zephyr/runs/6916504029?check_suite_focus=true

Also, check this https://static.sched.com/hosted_files/zephyr2022/a3/HOWTO_Get_Your_Zephyr_Patches_Merged_ZDS_2022.pdf, starting slide 21

File: dts/arm/st/f4/stm32f412.dtsi [hunk 2]
```diff
@@ -190,5 +190,18 @@
...
+
+
+		can1: can@40006400 {
+			compatible = "st,stm32-can";
+			reg = <0x40006400 0x400>;
+			interrupts = <19 0>, <20 0>, <21 0>, <22 0>;
+			interrupt-names = "TX", "RX0", "RX1", "SCE";
+			clocks = <&rcc STM32_CLOCK_BUS_APB1 0x02000000>;
+			status = "disabled";
+			label = "CAN_1";
```
Comment (FRASTM) [inline dts/arm/st/f4/stm32f412.dtsi:202]: 'label' is no more relevant

File: dts/arm/st/f4/stm32f412.dtsi [hunk 3]
```diff
@@ -190,5 +190,18 @@
 			status = "disabled";
 			label = "QUADSPI";
 		};
+
```
Comment (FRASTM) [inline dts/arm/st/f4/stm32f412.dtsi:193]: extra line to remove

**Review by FRASTM — COMMENTED** [review]
Also take this opportunity to add the` can2: can@40006800`

**Comment by pbonazzi** [comment]
> Also take this opportunity to add the` can2: can@40006800`

Continue conversation in #49411
